File: versão em python/2.py
# 2 - Questão:

import math


class QuantidadeVendedores:
    def __init__(self):
        self.intervalos = [[200, 299], [300, 399], [400, 499], [500, 599], [600, 699],
                           [700, 799], [800, 899], [900, 999], [1000, math.inf]]
        self.contadores = [0, 0, 0, 0, 0, 0, 0, 0, 0]

    def menu(self):
        opcao = -1
        while opcao != 0:
            print('\n- - - Selecione uma opção: - - -')
            print('1 - Para adicionar um salário de um vendedor')
            print('2 - Para ver a lista de contadores')
            print('0 - Sair')
            opcao = input('Digite a opção: ')

            if opcao == '1':
                self.adicionar_salario()
            elif opcao == '2':
                self.mostrar_lista()
            elif opcao == '0':
                break
            else:
                print('Selecione uma opção válida!')

    def adicionar_salario(self):
        salario = self.obter_salario()

        for i in range(len(self.contadores)):
            if salario >= self.intervalos[i][0] and salario <= self.intervalos[i][1]:
                self.contadores[i] += 1
        print('\nSalário adicionado!')

    def obter_salario(self):
        salario = 0
        while True:
            salario = int(input('Salário: '))
            if salario < 0:
                print('Digite um valor válido!')
            else:
                break
        return salario

    def mostrar_lista(self):
        opcao = ''
        while opcao != 's':
            print('\n--- Lista de Contadores: ---')
            print('$200 - $299:', self.contadores[0], 'vendedor(es),')
            print('$300 - $399:', self.contadores[1], 'vendedor(es),')
            print('$400 - $499:', self.contadores[2], 'vendedor(es),')
            print('$500 - $599:', self.contadores[3], 'vendedor(es),')
            print('$600 - $699:', self.contadores[4], 'vendedor(es),')
            print('$700 - $799:', self.contadores[5], 'vendedor(es),')
            print('$800 - $899:', self.contadores[6], 'vendedor(es),')
            print('$900 - $999:', self.contadores[7], 'vendedor(es),')
            print('$1000 em diante:', self.contadores[8], 'vendedor(es),')

            opcao = input('\nVoltar ao menu?(s/n): ').lower()
            if opcao == 's':
                break


if __name__ == "__main__":
    main = QuantidadeVendedores()
    main.menu()

File: 2.rb
# 2 - Questão:


class QuantidadeVendedores
    @@intervalos = [[200, 299], [300, 399], [400, 499], [500, 599], [600, 699],
                    [700, 799], [800, 899], [900, 999], [1000, Float::INFINITY]]
    @@contadores = [0, 0, 0, 0, 0, 0, 0, 0, 0]

    def menu
        opcao = -1
        while opcao != 0
            puts "\n- - - Selecione uma opção: - - -"
            puts "1 - Para adicionar um salário de um vendedor"
            puts "2 - Para ver a lista de contadores"
            puts "0 - Sair"

            print "Digite a opção: "
            opcao = gets.chomp

            if opcao == "1"
                adicionar_salario
            elsif opcao == "2"
                mostrar_lista
            elsif opcao == '0'
                break
            else
                puts "Selecione uma opção válida!"
            end
        end
    end

    def adicionar_salario
        salario = obter_salario

        for i in 0...@@contadores.length
            if salario >= @@intervalos[i][0] and salario <= @@intervalos[i][1]
                @@contadores[i] += 1
            end
        end
        puts "\nSalário adicionado!"
    end

    def obter_salario
        salario = 0
        while true
            print "Salário: "
            salario = gets.chomp.to_f

            if salario < 0
                puts "Digite um valor acima de zero!"
            else
                break
            end
        end
        return salario
    end

    def mostrar_lista
        opcao = ""
        while opcao != "s"
            puts "\n--- Lista de Contadores: ---"
            puts "$200 - $299: #{@@contadores[0]} vendedor(es),"
            puts "$300 - $399: #{@@contadores[1]} vendedor(es),"
            puts "$400 - $499: #{@@contadores[2]} vendedor(es),"
            puts "$500 - $599: #{@@contadores[3]} vendedor(es),"
            puts "$600 - $699: #{@@contadores[4]} vendedor(es),"
            puts "$700 - $799: #{@@contadores[5]} vendedor(es),"
            puts "$800 - $899: #{@@contadores[6]} vendedor(es),"
            puts "$900 - $999: #{@@contadores[7]} vendedor(es),"
            puts "$1000 em diante: #{@@contadores[8]} vendedor(es)."

            print "\nVoltar ao menu?(s/n): "
            opcao = gets.chomp.downcase
            if opcao == "s"
                break
            end
        end
    end
end


main = QuantidadeVendedores.new
main.menu
